src/ops/inc_multihead_self_attention.cu

```diff
@@ -1753,40 +1730,29 @@ void IncMultiHeadSelfAttention::inference_kernel_wrapper(
 
   // assert(input.data_type == weight.data_type);
   assert(input.data_type == output.data_type);
-  if (use_bias) {
-    assert(input.data_type == bias.data_type);
-  }
+  // if (use_bias) {
+  //   assert(input.data_type == bias.data_type);
+  // }
 
   if (input.data_type == DT_HALF) {
-    if (m->offload) {
-      pre_build_weight_kernel<half>(m, weight, input.data_type, stream);
-    }
-    half const *bias_ptr =
-        use_bias ? bias.get_half_ptr() : static_cast<half const *>(nullptr);
     Kernels::IncMultiHeadAttention::inference_kernel(
         m,
         bc,
         shard_id,
         input.get_half_ptr(),
-        m->offload ? static_cast<half *>(m->weight_ptr) : weight.get_half_ptr(),
+        static_cast<half const *>(nullptr), // weight_ptr is no longer used
```

Comment:
let's remove these null pointers and remove the weight/bias arguments to the inference_kernel